infra/lib/api-stack/fhir-converter-connector.ts

```diff
@@ -82,27 +86,31 @@ export function createQueueAndBucket({
       encryption: s3.BucketEncryption.S3_MANAGED,
     });
 
-  return { queue, dlq, bucket: fhirConverterBucket };
+  return { queue, dlq: dlq.queue, bucket: fhirConverterBucket };
 }
 
 export function createLambda({
+  lambdaLayers,
   envType,
   stack,
   vpc,
   sourceQueue,
   destinationQueue,
   dlq,
   fhirConverterBucket,
+  apiServiceDnsAddress,
   conversionResultQueueUrl,
   alarmSnsAction,
 }: {
+  lambdaLayers: ILayerVersion[];
   envType: EnvType;
   stack: Construct;
   vpc: IVpc;
-  sourceQueue: Queue;
-  destinationQueue: Queue;
-  dlq: DeadLetterQueue;
+  sourceQueue: IQueue;
+  destinationQueue: IQueue;
+  dlq: IQueue;
```

Comment:
Moving to interfaces increases decoupling.

Reply:
💯